### PR TITLE
Doc/ipython dep

### PR DESCRIPTION
**Pull request by tacaswell**
Add section to our minimum dependencies about IPython.

Closes matplotlib#16263

## PR checklist

- [x] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: doc/devel/min_dep_policy.rst [hunk 1]
```diff
@@ -51,6 +51,13 @@ versions no longer support our minimum NumPy or Python.
 
 We will work around bugs in our dependencies when practical.
 
+IPython and Matplotlib do not technically depend on each other, however there
```
Comment (timhoffm) [inline doc/devel/min_dep_policy.rst:54]: I was briefly confused why this is in "Python dependencies" (reading the dependency on Python). Can we rename the section "Python dependencies" -> "Other Python packages"

The current toc structures is

- Dependency version policy
  - Python and NumPy
  - Python dependencies

And I believe "Other Python packages" is more clear.

Comment (timhoffm) [inline doc/devel/min_dep_policy.rst:54]:
```suggestion
IPython and Matplotlib do not formally depend on each other, however there
```

File: doc/devel/min_dep_policy.rst [hunk 2]
```diff
@@ -51,6 +51,13 @@ versions no longer support our minimum NumPy or Python.
...
+is practical coupling between the projects.  There is no dependency to update,
```
Comment (timhoffm) [inline doc/devel/min_dep_policy.rst:55]:
```suggestion
is practical coupling between the projects via the matplotlib inline backend. 
There is no explicitly written dependency to update,
```
- We should at least hint at the reason for the coupling.
- "Dependency" is a bit vague. It's either the general term for coupling or the codified package relation.
  You want the latter. Is "explicitly written dependency"

File: doc/devel/min_dep_policy.rst [hunk 3]
```diff
@@ -51,6 +51,13 @@ versions no longer support our minimum NumPy or Python.
...
+but we will ensure the integration works with at least minor or major versions
+of IPython released in the 24 months prior to our planned release date.  We
```
Comment (timhoffm) [inline doc/devel/min_dep_policy.rst:57]:
```suggestion
of IPython released in the 24 months prior to our planned release date.
Matplotlib may or may not work with older versions and we
```

File: doc/devel/min_dep_policy.rst (hunk 2 above)
Comment (ianthomas23) [reply doc/devel/min_dep_policy.rst:55]: If you want to mention the reason for the coupling then it is not just via the Matplotlib inline backend, it is use of any Matplotlib backend with either IPython or any of the Jupyter subcommands.

Comment (timhoffm) [reply doc/devel/min_dep_policy.rst:55]: Yup, I just wanted to give an idea. I believe most people won't know what "use of any Matplotlib backend with either IPython or any of the Jupyter subcommands." means. Maybe (but still not fully correct) "via the ``%matplotlib`` command magic.

Comment (tacaswell) [reply doc/devel/min_dep_policy.rst:55]: I had written a sentence about event loops but decided that was too much detail.

Comment (timhoffm) [reply doc/devel/min_dep_policy.rst:55]: New suggestion
```suggestion
is practical coupling for the integration of Matplotlib into Ipython.
```

**Review by timhoffm — APPROVED** [review]

**Review by ianthomas23 — APPROVED** [review]

**Pull request merged**